### ISSUE TITLE
Dot-tilde (.~) broadcasting not supported in Turing models

### ISSUE THREAD

**Issue by NMUrban** [TuringLang/Turing.jl#1360]
It appears that broadcasting the `~` operator does not work.  For example:

```
z = randn(10000)

@model dottilde(y) = begin
    μ ~ Normal(0,1)
    σ ~ Gamma(1,1)
    y .~ Normal(μ,σ)
end

q = vi(dottilde(z), ADVI(1,2000))
```

results in an error:

```
MethodError: no method matching dot_tilde(::DynamicPPL.DefaultContext, ::DynamicPPL.SampleFromPrior, ::Normal{Float64}, ::Array{Float64,1}, ::Array{Float64,1}, ::DynamicPPL.ThreadSafeVarInfo{DynamicPPL.VarInfo{DynamicPPL.Metadata{Dict{DynamicPPL.VarName,Int64},Array{Distribution,1},Array{DynamicPPL.VarName,1},Array{Real,1},Array{Set{DynamicPPL.Selector},1}},Float64},Array{Base.RefValue{Float64},1}})
```

The code works if you replace `y .~ Normal(μ,σ)` with `y ~ filldist(Normal(μ,σ), length(y))`.

**Comment by krishvishal** [comment]
Try `y .~ Normal.(μ,σ)` instead of `y .~ Normal(μ,σ)`

**Comment by devmotion** [comment]
Looks like a duplicate of https://github.com/TuringLang/DynamicPPL.jl/issues/137 and should be fixed by https://github.com/TuringLang/DynamicPPL.jl/pull/147.

**Comment by krishvishal** [comment]
@devmotion I've been using `y .~ Normal.(μ,σ)`, can you tell me if this is correct?

**Comment by devmotion** [comment]
In the example above it shouldn't make any difference since `μ` and `σ` are just scalars. Of course, if that's not the case and, e.g., `μ` would be a vector with different values for each output, then one has to use `y .~ Normal.(μ, σ)`.

So I think both versions are "correct" in some way, but personally I would use `y .~ Normal(μ, σ)` in the example above (which currently unfortunately requires the bugfix mentioned above).

**Comment by devmotion** [comment]
Fixed in Turing 0.14 (which includes https://github.com/TuringLang/DynamicPPL.jl/pull/147).
